package.json

```diff
@@ -46,46 +52,38 @@
     "ws": "^7.5.10"
   },
   "devDependencies": {
-    "@babel/cli": "^7.16.0",
-    "@babel/core": "^7.16.0",
-    "@babel/eslint-parser": "^7.17.0",
-    "@babel/node": "^7.16.0",
-    "@babel/plugin-proposal-class-properties": "^7.16.0",
-    "@babel/plugin-proposal-object-rest-spread": "^7.16.0",
-    "@babel/plugin-transform-async-to-generator": "^7.16.0",
-    "@babel/plugin-transform-object-assign": "^7.16.0",
-    "@babel/plugin-transform-runtime": "^7.16.4",
-    "@babel/preset-env": "^7.16.4",
-    "@babel/preset-typescript": "^7.16.0",
-    "@babel/register": "^7.16.0",
+    "@babel/plugin-proposal-class-properties": "^7.18.6",
+    "@babel/plugin-proposal-object-rest-spread": "^7.20.7",
+    "@babel/plugin-transform-object-assign": "^7.25.9",
+    "@babel/plugin-transform-runtime": "^7.26.9",
+    "@babel/preset-env": "^7.26.9",
+    "@babel/preset-typescript": "^7.26.0",
+    "@babel/register": "^7.25.9",
```

Comment:
Unfortunately these are still needed for the Mocha to run properly, using different loader generated a TON of other issues I wasn't comfortable dealing with within this PR, updates should suffice for now.

Reply:
What issues did you face? We managed to easily solve this in GetStream/chat QA.

Reply:
After installing `ts-node` and replacing the loader (through the `--require` Mocha option) you're immediately after running greeted with `Exception during run: Error: Cannot require() ES Module` - if you rename `.js` to `.mjs` you get something else and after that it has issue with our tests importing from `src` at which point I stopped pursuing.

Reply:
The QA seems to have all the tests in TS, the tests here are a combination of both - at some point I was able to run tests (with some different loader, can't remember now) but the TS was failing on completely bogus errors which are not detected normally.

Reply:
Let me check one last time with separate `tsconfig` for testing only.

Reply:
[baa9758](https://github.com/GetStream/stream-chat-js/pull/1468/commits/baa97582fb75a72eded9090bdfd2af841e2048d3)

![](https://media1.tenor.com/m/Cyr2PR6E3kkAAAAC/sacha-baron.gif)